### PR TITLE
Implement parallization. Process multiple cases

### PR DESCRIPTION
**Pull request by ssskkkky**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -28,6 +28,7 @@ app.disable('x-powered-by');
 app.listen(port);
 
 console.log(`Server is running at http://127.0.0.1:${port}`);
+console.log(`Server is running`,host_dir);
```
Comment (12ff54e) [inline server.js:31]: The meaning of this logging message seems confusing.

**Review by 12ff54e — CHANGES_REQUESTED** [review]
Check individual code comments pls

File: server.js (hunk 1 above)
Comment (ssskkkky) [reply server.js]: This line was deleted

File: server.js [hunk 2]
```diff
@@ -45,19 +46,29 @@ app.post('/', async (req, res) => {
...
             decodeURI(req.body.gtc_output)
         );
-        output = new GTCOutput(GTC_outputDir);
-        console.log(`path set to ${GTC_outputDir}`);
-
-        await output.getSnapshotFileList();
-        await output.check_tracking();
+        let currentOutput;
+        if (output[req.body.gtc_output] === undefined) {
+            currentOutput = output[req.body.gtc_output] = new GTCOutput(
```
Comment (12ff54e) [inline server.js:51]: Beware that `req.body.gtc_output` is percent-encoded using `encodeURI`.

File: server.js [hunk 3]
```diff
@@ -99,12 +110,12 @@ app.get('/plotType/:type', async (req, res) => {
...
 
 app.get('/Summary', (req, res) => {
-    generateSummary().then(res.json.bind(res));
+    generateSummary(output[req.query.dir]).then(res.json.bind(res));
 });
 
 app.get('/data/basicParameters', (req, res) => {
-    output.read_para().then(() => {
-        res.json(output.parameters);
+    output[req.query.dir].read_para().then(() => {
```
Comment (12ff54e) [inline server.js:117]: The default behaviour of query string parser will decode percent-encoded strings. Inconsistency occurs when the path contains special characters need to be encoded, e.g. '\\'. `encodeURI` this string should resolve the problem.

**Review by 12ff54e — CHANGES_REQUESTED** [review]
Bug found when testing in Windows, check the comment pls.

File: server.js (hunk 3 above)
Comment (ssskkkky) [reply server.js]: This bug was fixed, please try it on a windows

**Pull request merged**